specifyweb/frontend/js_src/lib/components/WorkBench/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { HotTable } from '@handsontable/react';
+
+import { useAsyncState } from '../../hooks/useAsyncState';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { useErrorContext } from '../../hooks/useErrorContext';
+import { commonText } from '../../localization/common';
+import { wbText } from '../../localization/workbench';
+import { ajax } from '../../utils/ajax';
+import { f } from '../../utils/functools';
+import type { GetSet } from '../../utils/types';
+import { className } from '../Atoms/className';
+import { LoadingContext } from '../Core/Contexts';
+import { LoadingScreen } from '../Molecules/Dialog';
+import { useMenuItem } from '../Header/MenuContext';
+import { treeRanksPromise } from '../InitialContext/treeRanks';
+import { Dialog } from '../Molecules/Dialog';
+import { NotFoundView } from '../Router/NotFoundView';
+import type { Dataset } from '../WbPlanView/Wrapped';
+import { WbViewReact } from './WbView';
+
+const fetchTreeRanks = async (): Promise<true> => treeRanksPromise.then(f.true);
+
+const fetchDataSet = async (
+  dataSetId: number | undefined
+): Promise<Dataset | undefined> =>
+  typeof dataSetId === 'number'
+    ? ajax<Dataset>(`/api/workbench/dataset/${dataSetId}/`, {
+        headers: { Accept: 'application/json' },
+      }).then(({ data }) => data)
+    : undefined;
+
+// BUG: intercept 403 (if dataset has been transferred to another user)
+function useDataSet(
+  dataSetId: number | undefined
+): GetSet<Dataset | undefined> {
+  return useAsyncState(
+    React.useCallback(async () => fetchDataSet(dataSetId), [dataSetId]),
+    true
+  );
+}
+
+export function WorkBenchReact(): JSX.Element {
```

Comment:
every component is using react so having `React` in the name is a bit redundant...

if needed, we usually call such components `WorkBenchWrapper`, `WorkBenchContainer` or `WorkBenchParent` instead